src/pages/IndivPlaylistPage.js

```diff
@@ -0,0 +1,129 @@
+        try {
+            const updated = playlist["songs"].filter((song, i) => {
+                return i !== index
+            });
+            let newPlaylist = playlist;
+            newPlaylist["songs"] = updated;
+            const response = await axios.put(
+                'http://localhost:5000/playlists/' + newPlaylist['_id'].toString(), newPlaylist
+            );
+            if (response && response.status === 204) {
+                setPlaylist({
+                    "title": playlistName,
+                    "songs": playlist["songs"],
+                    "_id": playlist['_id']
+                });
+            }
+        }
+        catch (error){
+            //We're not handling errors. Just logging into the console.
+            console.log(error);
+            return false;
+        }
+    }
+
+    async function findAndAddSong(song) {
+        const songTitle = song.val;
+        try {
+            const result = await axios.get("/auth/search/" + songTitle);
+            if (result !== undefined) {
+                const songToAdd = result.data["tracks"]["items"][0];
+                updateList(songToAdd);
+            }
+            else {
+                console.log("Not found.");
+            }
+        }
+        catch(error) {
+            console.log("I'M A FAILURE");
+            console.log(error);
+        }
+    }
+
+    function updateList(song) {
+        makePostCall(song).then( result => {
+            if (result && result.status === 204) {
+                setPlaylist({
+                    "title": playlistName,
+                    "songs": playlist["songs"],
+                    "_id": playlist['_id']
+                });
+                // let newPlaylist = playlist;
+                // newPlaylist["songs"] = [...newPlaylist["songs"], song];
+                // setPlaylist(newPlaylist);
+            }
+        });
+    }
+
+    async function makePostCall(song){
+        try {
+            let newPlaylist = playlist;
+            newPlaylist["songs"] = [...newPlaylist["songs"], song];
+            const response = await axios.put(
+                'http://localhost:5000/playlists/' + newPlaylist['_id'].toString(), newPlaylist
+            );
+            return response;
+        }
+        catch (error) {
+            console.log(error);
+            return false;
+        }
+    }
+
+    async function fetchAll(){
+        try {
+            const response = await axios.get('http://localhost:5000/playlists?title=' + playlistName);
+            return response.data[0];
+        }
+        catch (error){
+            //We're not handling errors. Just logging into the console.
+            console.log(error);
+            return false;
+        }
+    }
+
+    useEffect(() => {
+        fetchAll().then( result => {
+            if (result)
+                setPlaylist(result);
+        });
+    }, [] );
+
+
+    const PlaylistHeader = () => {
+        return <h1>{playlistName}</h1>;
+    }

```

Comment:
Do we need a const for a tag?

Reply:
no but it looks a bit cleaner to me on the return value